src/System.Reflection.MetadataLoadContext/src/System/Reflection/PathAssemblyResolver.cs

```diff
@@ -56,63 +54,43 @@ public PathAssemblyResolver(IEnumerable<string> assemblyPaths)
 
         public override Assembly Resolve(MetadataLoadContext context, AssemblyName assemblyName)
         {
-            // On the first access to this assembly name, load each assembly from the provided path(s).
-            LoadMatchingAssembliesFromPaths(context, assemblyName.Name);
-
-            Assembly candidate = null;
-
-            foreach (Assembly a in context.GetAssemblies())
-            {
-                if (assemblyName.Name.Equals(a.GetName().Name, StringComparison.OrdinalIgnoreCase) && IsPublicKeyTokenCompatible(assemblyName, a))
-                {
-                    // Pick the highest version.
-                    if (candidate == null || NormalizeVersion(a.GetName().Version) > NormalizeVersion(candidate.GetName().Version))
-                    {
-                        candidate = a;
-                    }
-                }
-            }
-
-            return candidate;
-        }
-
-        private void LoadMatchingAssembliesFromPaths(MetadataLoadContext context, string simpleName)
-        {
+            Assembly candidateWithSamePkt = null;
+            Assembly candidateIgnoringPkt = null;
             List<string> paths;
-            if (_fileToPaths.TryGetValue(simpleName, out paths))
+            if (_fileToPaths.TryGetValue(assemblyName.Name, out paths))
             {
+                byte[] pktFromName = assemblyName.GetPublicKeyToken() ?? Array.Empty<byte>();
```

Comment:
Nit: if you type these locals as `ReadOnlySpan<byte>` instead of byte[], then you'd get an implicit conversion, and wouldn't have to cast elsewhere. I think that would be a little easier to read.

Reply:
OK done